dialogs/primitives.py

```diff
@@ -1,12 +1,18 @@
 from typing import List
+from functools import wraps
 
 from .types import Dialog, DialogGenerator, ClientResponse, RunSubdialog, DialogState
 
 
+def primitive(dialog):
```

Comment:
Can any object be a primitive or only functions? If the latter I would add a type signature.
Also, I'm sure I'll find out as I read on but what does primitive do?

Reply:
It's a decorator. I'm looking for a way to reduce the number of parameters a dialog takes, it's getting to 4 (run, state, client_response, send), but 99% of dialogs only need the first one. This decorator affects which args get injected in